### PR TITLE
Fix RangesHelper.sanitize_ranges for empty list

### PR DESCRIPTION
**Pull request by Qwerty5Uiop**
## Motivation

For empty list `RangesHelper.sanitize_ranges/1` returns `[nil]`, also it works incorrect for ranges with `-1` step

## Summary by CodeRabbit

* **New Features**
  * Range parsing now supports a trailing "latest" marker and adds a helper to get the minimum block from a range string.

* **Bug Fixes**
  * Improved merging/sanitization of overlapping and adjacent ranges; preserves ascending/descending direction, ordering, and is more nil-safe.

* **Refactor**
  * Centralized parsing logic into a shared utility and reused by other components.

* **Tests**
  * Added tests covering parsing, merging, ordering, edge cases, and empty input.

### CODE REVIEW

File: apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/utility/ranges_helper.ex [hunk 1]
```diff
@@ -136,39 +160,11 @@ defmodule EthereumJSONRPC.Utility.RangesHelper do
   @doc """
   Rejects empty ranges and merges adjacent ranges
   """
-  @spec sanitize_ranges([Range.t() | integer()]) :: [Range.t() | integer()]
+  @spec sanitize_ranges([Range.t() | nil]) :: [Range.t()]
```
[the remark this answers is not in the document]
Comment (Qwerty5Uiop) [reply apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/utility/ranges_helper.ex:163]: It can't return integers and the is no `to_latest` variable inside
